src/workflows/airqo_etl_utils/data_validator.py

```
@@ -120,14 +122,15 @@ def remove_outliers(data: pd.DataFrame) -> pd.DataFrame:
-        for col in validated_columns:
-            mapped_name = Config.AIRQO_DATA_COLUMN_NAME_MAPPING.get(col, None)
-            data[col] = data[col].apply(
-                lambda x: DataValidationUtils.get_valid_value(
-                    column_name=mapped_name, row_value=x
+        if remove_outliers:
+            validated_columns = list(chain.from_iterable(filtered_columns.values()))
+            for col in validated_columns:
+                mapped_name = Config.AIRQO_DATA_COLUMN_NAME_MAPPING.get(col, None)
+                data[col] = data[col].apply(
+                    lambda x: DataValidationUtils.get_valid_value(
+                        column_name=mapped_name, row_value=x
```

Ruff (0.8.2)

Function definition does not bind loop variable `mapped_name`

(B023)